### DISCUSSION TITLE
Simulate Times to Failure from Right Censored data

### DISCUSSION THREAD

**Discussion by Konrad-H** [CamDavidsonPilon/lifelines#1260] [Q&A] [answered]
Is there a way to simulate random times until failure event from a fitted distribution taking into account the time it already passed? 

I couldn't find it in the documentation. Thank you!

**Comment by CamDavidsonPilon** [comment, marked as answer]
Hi @Konrad-H - there is something that you can use.  Give a fitted univariate model `f`, you can provide uniform random variates to the `percentile` function:
```
wf = WeibullFitter()
T = np.random.exponential(4, size=1000)
wf.fit(T)
sim = wf.percentile(np.random.random(size=10000)) # simulate 10000 survival times
```
This works due to [inverse transform sampling](https://en.wikipedia.org/wiki/Inverse_transform_sampling).

Now, you want to take into account time that has already passed, call this time _s_. For this, you just need to remove any entries below _s_ in the resulting vector. 
```
sim[sim > s]
```

**Reply by Konrad-H** [reply]
Hi! 
Really thank you for your answer! 
I thought about your answer and something doesnt quite add up to me.
By the definition of conditional probability $P(T<x | T>t) = P(T<x & T>t)/P(T>t)$.
Anyway, the (bootstrapped) fit parameters and histograms are the same for the Conditional method so I believe there has to be an error in my thought process.

**Reply by CamDavidsonPilon** [reply]
Take a look at this answer first for a derivation of the conditional survival function: https://stats.stackexchange.com/questions/519773/remaining-useful-life-for-weibull-distributed-lifetimes/519902#519902

Point is that the conditional survival function is just a rescaled version of the original survival function, so sampling from the conditional should be identical to sampling from the original.

**Reply by Konrad-H** [reply]
Ok! Everything is crystal clear now! 🤩
